### ISSUE TITLE
Support Ubuntu 22.04 LTS (Jammy Jellyfish)

### ISSUE THREAD

**Issue by Bengt** [RadeonOpenCompute/ROCm#1730]
Ubuntu 22.04 LTS (Jammy Jellyfish) is the current LTS version of Ubuntu.

https://ubuntu.com/blog/ubuntu-22-04-lts-released

@ROCmSupport repeatedly promised support it once it is out:

https://github.com/RadeonOpenCompute/ROCm/issues/1590#issuecomment-942093569
https://github.com/RadeonOpenCompute/ROCm/issues/1612#issuecomment-965049119

Please fulfill that promise by stating official support in documentation, and also by providing an official release of ROCm that works with this version of Ubuntu.

The absence of a working release is already leading to issues:

https://github.com/RadeonOpenCompute/ROCm/issues/1713

**Comment by Bengt** [comment]
I am such a researcher, stuck with an outdated OS. :/

**Comment by keryell** [comment]
@BloodyIron I will ask internally.

**Comment by BloodyIron** [comment]
@keryell I really hope that AMD takes it seriously :/ Thanks for asking internally :)

**Comment by Bengt** [comment]
For future reference:

https://developer.nvidia.com/blog/nvidia-releases-open-source-gpu-kernel-modules/

The situation on the Nvidia side is quite complex, with closed-source firmware and user space software. However, by releasing the kernel module as open source, they enabled lots of possible improvements, like DMA-BUF. Note, that Nvidia did not release their compute component CUDA as open source, which is equivalent to AMD's ROCm, we are concerned about here. So technically, AMD is still ahead in terms of open-source support, which is, of course, the best way of being ahead. Still, in order for them to stay relevant, AMD urgently needs to release their ROCm packages in a version that is compatible with relevant operating systems. Every day that goes by is an incentive to turn the back on AMD products and in parallel Nvidia is ramping support for theirs.

**Comment by DrBlackross** [comment]
Kinda, beating a dead horse kinda question....

But, how do you jump over 22.04 to 22.10 on https://repo.radeon.com/amdgpu-install/ ? Was reading the pdf from the rocm 5.11 tarball release, been trying to fix opencl with rocm (since i updated to 22.04, lol, decades later still same issues with linux and graphic cards).

Sidenote: It wants to install on 22.04 with amdgpu-install, just can't find amdgpu-lib, amdgpu-lib32, rocm-opencl-runtime, rocm-hip-runtime... just don't want to reinstall or downgrade. Like i said before, decades of this, make menuconfig and compile it in the kernel would be nice.

**Comment by Bengt** [comment]
@DrBlackross Your request for ROCm to support Ubuntu 22.10 is off-topic in this issue about 22.04. Please open a separate issue about that.

**Comment by lionelchauvin** [comment]
@Bengt DrBlackross talks about Radeon™ Software for Linux® 22.10 on Ubuntu 22.04, not Ubuntu 22.10
Yes, this number version is confusing.

**Comment by Bengt** [comment]
@lionelchauvin, ah, I see. Thanks for the clarification. Is this repository the right place to report issues about AMD/Radeon Software for Linux?

**Comment by lionelchauvin** [comment]
@Bengt Perhaps I misunderstood too and he was talking about amdgpu 22.10.

Anyway I don't understand why Radeon™ Software for Linux® 22.20 (that contains rocm) has been released without the support of the most popular linux distribution. (https://www.amd.com/en/support/kb/release-notes/rn-amdgpu-unified-linux-20-20)

**Comment by keryell** [comment]
> Anyway I don't understand why Radeon™ Software for Linux® 22.10.2 (that contains rocm) has been released without the support of the most popular linux distribution.

@lionelchauvin I guess because it targets only the HPC  and datacenter ML niches?

**Comment by keryell** [comment]
> I really hope that AMD takes it seriously :/ Thanks for asking internally :)

@BloodyIron following the recent GitHub discussions, a meeting was organized last week with 19 AMD employees attending. So, some people care and I can see more activity on GitHub now.
Let's see how far this flies... :-)

**Comment by lionelchauvin** [comment]
> > Anyway I don't understand why Radeon™ Software for Linux® 22.10.2 (that contains rocm) has been released without the support of the most popular linux distribution.
> 
> @lionelchauvin I guess because it targets only the HPC and datacenter ML niches?

I doubt it is the unique reason because Amdvlk is supposed to target customers and has the same treatment.

**Comment by keryell** [comment]
> I doubt it is the unique reason because Amdvlk is supposed to target customers and has the same treatment.

@lionelchauvin Just looking at the description of https://github.com/RadeonOpenCompute/ROCm it says *ROCm - Open Source Platform for HPC and Ultrascale GPU Computing*. But I hope it should be updated to work also on machines cheaper than $10M. ;-)

**Comment by BloodyIron** [comment]
Yes, and ROCm is the only way to get OpenCL for consumer GPUs while using the AMDGPU driver, which is the preferred driver for gaming, as AMDGPU-Pro has worse performance for gaming. So while that may be what it says _now_, the ROCm suite has been the way to get OpenCL for years now for consumer GPUs when using AMDGPU. So I don't frankly care that it says it's for "HPC and Ultrascale GPU" without mentioning consumer usage. If AMD is not going to expose OpenCL in a way that works with AMDGPU drivers then they should frankly fully remove the declaration of OpenCL on the box for consumer GPUs considering Linux (and gaming on Linux) is a supported platform and use, supposedly.

**Comment by roblem** [comment]
Another researcher (not gamer) that is waiting for 22.04 support.  I have 2 Radeon VII's that [**will not work**](https://github.com/RadeonOpenCompute/ROCm/issues/1431) with upstream radeon driver and Rocm unless the kernel/upstream kmods is fairly recent.  This more recent kernel is not available under 20.04.  It is available for 22.04.  So this support is required for operability of hardware, and downgrading to 20.04 won't solve the issue.

**Comment by B0tBuilder** [comment]
I feel your pain.  Unfortunately, it appears that support for the most recent Ubuntu LTS is not a priority.

**Comment by BloodyIron** [comment]
I guess delivering on what is advertised on the box is not a priority...

**Comment by Bengt** [comment]
Are GPUs sold with Ubuntu LTS support advertised on the box? Can you please share?

**Comment by B0tBuilder** [comment]
@Bengt Wouldn't that be amazing!  don'tclaims support (on their website) for certain Ubuntu LTS releases and makes no specific claims regarding future support for new releases (to be clear).  Though one would conclude that if you are intent on supporting Linux, it may be a good idea to get ahead of the curve with respect to what is probably the most widely used Linux Distro (if you include flavors).

This is not an issue of some sort of false advertising as some seem to be claiming.  Just a lack of desire to commit adequate resources.  When your competitor dominates the field (for AI) it is probably most profitable to invest more resources in design wins for large deployments that use specialized environments.  Unless more funds are allocated, this necessarily results in fewer resources assigned to supporting the newer OS releases used by a smaller group of researchers and enthusiasts.  Oh... and crypto miners, but for obvious reasons, I feel no sympathy for them whatsoever.

Regardless of the reasons, if you dont like AMD's support and are unhappy with it, there is an obvious solution and alternative.

**Comment by BloodyIron** [comment]
1. OpenCL 2.1 is on the Wikipedia page talking about capabilities (somehow not found on AMD's website it seems) : https://en.wikipedia.org/wiki/Radeon_RX_6000_series
2. AMD's website for RX 6600, for example, lists "OS Support Windows 11 - 64-Bit Edition Windows 10 - 64-Bit Edition Linux x86_64" : https://www.amd.com/en/products/graphics/amd-radeon-rx-6600

It's a reasonable thing to expect OpenCL to work on the most popular Linux distro for a device that is marketed and designed for gaming + productivity.

**Comment by B0tBuilder** [comment]
@

> 1. OpenCL 2.1 is on the Wikipedia page talking about capabilities (somehow not found on AMD's website it seems) : https://en.wikipedia.org/wiki/Radeon_RX_6000_series
> 2. AMD's website for RX 6600, for example, lists "OS Support Windows 11 - 64-Bit Edition Windows 10 - 64-Bit Edition Linux x86_64" : https://www.amd.com/en/products/graphics/amd-radeon-rx-6600
> 
> It's a reasonable thing to expect OpenCL to work on the most popular Linux distro for a device that is marketed and designed for gaming + productivity.

Agreed, but that is not what is happening and AMD has made no commitment to do so.

**Comment by BloodyIron** [comment]
Thanks for your efforts @cgmb ! Here's hoping this all gets sorted please!

**Comment by DrBlackross** [comment]
Might want to tell people to put their username in the  '  ' (instead of yours lol)

> Permissions (change username)

**Comment by DrBlackross** [comment]
Ok, i rebooted.... downloaded lolminer as a lark (just because i know it'll pull the opencl devices)

~/Desktop/1.51a$ sudo ./lolMiner --list-devices
OpenCL driver detected. Number of OpenCL supported GPUs: 0 
No Cuda driver or GPUs detected. 

hmmm, didn't work, BUT, it installed rocm.

now, mmpos usb bootable miner (i boot to it when im out of country) will detect and load the opencl drivers without a problem and mine whatever i want with opencl

(god only knows what linux distro is in there)

i know its probably me, i'm missing something

**Comment by DrBlackross** [comment]
oops, forgot to add this

~/Desktop/xmrig$ ./xmrig --print-platforms
Number of platforms:        1

  Index:                    0
  Profile:                  FULL_PROFILE
  Version:                  OpenCL 2.1 AMD-APP (3423.0)
  Name:                     AMD Accelerated Parallel Processing
  Vendor:                   Advanced Micro Devices, Inc.
  Extensions:               cl_khr_icd cl_amd_event_callback

and from mmpos...

 v3.0.16
Agent version
5.10.83-mmp / 20.04
Kernel version
5.13.18 / 21.50.2
AMD driver versions
1.0.29
Addin version
510.73.05
Nvidia driver version
stable / 3.0.4 - (castle)
System version

**Comment by Espionage724** [comment]
> Ok, i rebooted.... downloaded lolminer as a lark (just because i know it'll pull the opencl devices)

I use [nsfminer](https://github.com/no-fee-ethereum-mining/nsfminer) and it seems to work fine with ROCm OpenCL. I tried `lolminer --list-devices` but it does a Segmentation fault (may be unrelated to ROCm).

**Comment by DrBlackross** [comment]
00:35:51 miner nsfminer 1.3.14 (No stinkin' fees edition)
00:35:51 miner Copyright 2021 Jean M. Cyr, Licensed under the terms
00:35:51 miner  of the GNU General Public License Version 3
00:35:51 miner https://github.com/no-fee-ethereum-mining/nsfminer
00:35:51 miner Build: linux/release/gnu
00:35:51 miner 3rd Party: GCC 10.2.0, Boost 1.76.0
00:35:51 miner 3rd Party: OpenSSL 1.1.1j  16 Feb 2021
00:35:51 miner Running as user: drblackross

Error: No usable mining devices found
even as 'root'.... nope

**Comment by B0tBuilder** [comment]
I suspect this is unrelated to the topic at hand and should probably be moved to another thread.

**Comment by DrBlackross** [comment]
k

**Comment by SciPyPanda** [comment]
Is this issue being solved?

**Comment by cgmb** [comment]
@SciPyPanda, yes. There was a solution found. It's a work in progress.

**Comment by rajhlinux** [comment]
I have AMD Radeon RX580 GPU, how can I install "amdgpu-pro" drivers to encode H.264 using FFMPEG using hardware acceleration from AMD AMF on Ubuntu 22.04?

Thanks.

**Comment by My1** [comment]
> Does AMD even care about OpenCL on consumer GPUs on Linux? You know, the feature that is literally sold on the box? Because I have seen zero evidence that AMD has interest in providing OpenCL working on Linux for consumer GPUs since like ROCm v3.3 for RX 580-era GPUs.

even if not the amdgpu-install script currently depends on rocm so that kinda sux

**Comment by ableeker** [comment]
The Ubuntu versions of amdgpu-install 22.20, and 22.20.1 come in a folder jammy, so you may assume these will support Ubuntu 2204.

I then actually try to use it to install HIP like so:

`amdgpu-install --usecase=hip,hiplibsdk`

This fails with the following error:
```
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 rocm-llvm : Depends: python but it is not installable
             Depends: libstdc++-5-dev but it is not installable or
                      libstdc++-7-dev but it is not installable
             Depends: libgcc-5-dev but it is not installable or
                      libgcc-7-dev but it is not installable
E: Unable to correct problems, you have held broken packages.
```
Ubuntu 2204 doesn't offer these packages anymore, only later versions. For HIP anyway, they're not even used. I install dummy packages just to satisfy the dependencies, and HIP install and runs fine. That of course could mean that other ROCM parts fail if they need the packages. libgcc-dev, and libstdc++-dev may well be fine with later versions, but I think that Python refers to Python 2, and if that's correct, some parts will most likely fail if they try to use Python 3 instead.

**Comment by jlublin** [comment]
It seems the driver packages does not actually need python, libstdc++-7-dev or libgcc-7-dev if appropriate newer packages are available. Here is a fairly simple workaround by creating a dummy package:

1. Install "equivs"
2. Run "equivs-control amdgpu-driver-fixes"
3. Edit the new amdgpu-fixes file, you may use the following lines

```
Package: amdgpu-driver-fixes
Provides: python,libstdc++-7-dev,libgcc-7-dev
Architecture: all
Description: Fixes the AMD GPU driver installation on Ubuntu 22.04
```
4. Run "equivs-build amdgpu-driver-fixes"
5. Install the new amdgpu-driver-fixes_1.0_all.deb

If you have python-is-python3 then you need to uninstall it since it will create a conflict.
I have not tested all functionallity but HIP in Blender works fine now.

**Comment by Fuzzillogic** [comment]
@ableeker @jlublin THANK YOU! I bought an RX 6800XT in September 2021, expecting I could use it soon with Blender, which would be my main use case. I've been using AMD GPUs since Matrox quit the market, and I am Ubuntu user since 2012. Thus far AMD had been a good to excellent choice. So, I'm a bit miffed, to say the least, about the current situation with HIP on Ubuntu 22.04 LTS.

But thanks to you, almost a year later, it now *finally* works in Blender. A scene I made earlier took 6 minutes on CPU (Ryzen 2700X. Yes, that's AMD as well). It now takes 38 SECONDS on GPU, including some CPU-time for denoising. 

And still the hardware ray tracing capabilities are unused :(

I guess as mere enthusiast / hobbyist I'm not the prime market target for AMD. Still, letting us dangle for so long doesn't sit well with me.

**Comment by mrkapqa** [comment]
what was helping in my case was to use this here

https://amdgpu-install.readthedocs.io/en/latest/install-installing.html#installing-the-all-open-use-case

**amdgpu-install -y --usecase=graphics
**

so it would install, when all other options failed on Ubuntu 22.04 with newer Amdgpu 22.20 repository for legacy graphics card Radeon RX 550X (Polaris)

**Comment by sosheskaz** [comment]
I'm not too knowledgeable about the subject matter, but I arrived here because I wanted to play video games after updating to 22.04.

I tried the above, but still had performance issues. After also installing the `opencl` and `workstation` usecases, I am seeing good performance. I am not sure which one did the trick.

**Comment by rajkhand** [comment]
I used this method to install the AMD driver
[https://askubuntu.com/questions/1417418/unmet-dependencies-ubuntu-22-04-amdgpu-hip-support](url)
There were no errors but not sure if this is correct

**Comment by ReeceSX** [comment]
Linux bros, did they lie to us about the open soros drivers and stable userland? Why does the distro LTS version matter for a compute ICD? LOL

**Comment by melt7777** [comment]
Yes, AMD sold us out... we have not been able to get opencl working on any AMD on Ubuntu version newer than 18.04 LTS.

**Comment by Bengt** [comment]
Building upon the answer by @jlublin, here is my procedure:

0. Make sure you do not have `python-is-python3` installed:  
   ```sudo apt remove --yes python-is-python3```
1. Make sure python, libstdc++-7-dev or libgcc-7-dev are installed:  
    ```sudo apt install --yes python3 libgcc-11-dev libstdc++-11-dev```
2. Install the "equivs" package using APT:  
    ```sudo apt install --yes equivs```
3. Create a file named `amdgpu-dependencies` containing the package metadata:  
   ```printf "Package: amdgpu-dependencies\nProvides: python,libstdc++-7-dev,libgcc-7-dev\nArchitecture: all\nDescription: Fixes the AMD GPU installation on Ubuntu 22.04\n" > amdgpu-dependencies```
4. Create a Debian package from the package metadata:
    ```equivs-build amdgpu-dependencies```
5. Install the package providing the dependencies:
    ```sudo dpkg -i amdgpu-dependencies_1.0_all.deb```
6. Cleanup leftover files:  
   ```rm amdgpu-dependencies*```

To verify this has worked, install and run the ```amdgpu``` installation script:

0. Add yourself to the `video` user group:  
   ```sudo usermod -a -G video $LOGNAME```
1. Download the installation script package:  
   ```wget https://repo.radeon.com/amdgpu-install/22.20.3/ubuntu/jammy/amdgpu-install_22.20.50203-1_all.deb```
2. Install the installation script package:  
   ```sudo apt-get install --yes ./amdgpu-install_22.20.50203-1_all.deb```
3. Clean up the left-over package file:
   ```rm ./amdgpu-install_22.20.50203-1_all.deb```
4. Run the installation script:  
    ```amdgpu-install --help```

**Comment by zzyjsjcom** [comment]
**Add focal source to sources.list can fix libstdc++-7-dev issue**

Add the follow line to the file: /etc/apt/sources.list
deb [arch=amd64] http://archive.ubuntu.com/ubuntu focal main universe

**Test pass on AMD Radeon RX 6500 XT, using the following instruction.**
# for ubuntu-22.04
// AMD Radeon RX 6500 XT
wget -c https://repo.radeon.com/amdgpu-install/22.20.3/ubuntu/jammy/amdgpu-install_22.20.50203-1_all.deb
sudo apt-get install ./amdgpu-install_22.20.50203-1_all.deb
sudo apt-get update
amdgpu-install --opencl=rocr

My full sources.list:

deb https://mirrors.tuna.tsinghua.edu.cn/ubuntu/ jammy main restricted universe multiverse
# deb-src https://mirrors.tuna.tsinghua.edu.cn/ubuntu/ jammy main restricted universe multiverse
deb https://mirrors.tuna.tsinghua.edu.cn/ubuntu/ jammy-updates main restricted universe multiverse
# deb-src https://mirrors.tuna.tsinghua.edu.cn/ubuntu/ jammy-updates main restricted universe multiverse
deb https://mirrors.tuna.tsinghua.edu.cn/ubuntu/ jammy-backports main restricted universe multiverse
# deb-src https://mirrors.tuna.tsinghua.edu.cn/ubuntu/ jammy-backports main restricted universe multiverse
deb https://mirrors.tuna.tsinghua.edu.cn/ubuntu/ jammy-security main restricted universe multiverse
# deb-src https://mirrors.tuna.tsinghua.edu.cn/ubuntu/ jammy-security main restricted universe multiverse

# deb https://mirrors.tuna.tsinghua.edu.cn/ubuntu/ jammy-proposed main restricted universe multiverse
# deb-src https://mirrors.tuna.tsinghua.edu.cn/ubuntu/ jammy-proposed main restricted universe multiverse

deb [arch=amd64] http://archive.ubuntu.com/ubuntu focal main universe

Logs:
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy
$ clinfo
Number of platforms                               1
  Platform Name                                   AMD Accelerated Parallel Processing
  Platform Vendor                                 Advanced Micro Devices, Inc.
  Platform Version                                OpenCL 2.1 AMD-APP (3452.0)
  Platform Profile                                FULL_PROFILE
  Platform Extensions                             cl_khr_icd cl_amd_event_callback
  Platform Extensions function suffix             AMD
  Platform Host timer resolution                  1ns

  Platform Name                                   AMD Accelerated Parallel Processing
Number of devices                                 1
  Device Name                                     gfx1034
  Device Vendor                                   Advanced Micro Devices, Inc.
  Device Vendor ID                                0x1002
  Device Version                                  OpenCL 2.0
  Driver Version                                  3452.0 (HSA1.1,LC)
  Device OpenCL C Version                         OpenCL C 2.0
  Device Type                                     GPU
  Device Board Name (AMD)                         AMD Radeon RX 6500 XT
  Device PCI-e ID (AMD)                           0x743f
  Device Topology (AMD)                           PCI-E, 0000:03:00.0
  Device Profile                                  FULL_PROFILE

After fix this issue, you can remove the following line from sources.list file:
deb [arch=amd64] http://archive.ubuntu.com/ubuntu focal main universe

**Comment by xuhuisheng** [comment]
ROCm-5.3 provided official support for Ubuntu-22.04 (jammy).
<https://docs.amd.com/bundle/ROCm-Release-Notes-v5.3/page/About_This_Document.html>

**Comment by AbelVM** [comment]
> ROCm-5.3 provided official support for Ubuntu-22.04 (jammy). https://docs.amd.com/bundle/ROCm-Release-Notes-v5.3/page/About_This_Document.html

IT'S WORKING FLAWLESSLY! :tada: 

(but you should run `apt update && apt full-upgrade` afterwards, as they have published newer packages of almost everything right after the release)

**Comment by Chase-san** [comment]
@AbelVM So how did you install it? There is no release for v5.3 yet.

**Comment by mrplow** [comment]
> @AbelVM So how did you install it? There is no release for v5.3 yet.

Sure there is, did you follow the link?

Finally I can unsub from this thread!

**Comment by Chase-san** [comment]
It's release notes, which are great, but it does not provide a direct link to the downloads or installation guide. If anyone else sees this, go here instead https://docs.amd.com/bundle/ROCm-Downloads-Guide-v5.3/page/Introduction_to_AMD_ROCm_Installation_Downloads_Guide_for_Linux.html

**Comment by Rmalavally** [comment]
@Chase-san 

Thank you for your feedback. The ROCm Release Notes document for the ROCm v5.3 release is now updated with a link to the ROCm Downloads Guide. You can access the latest version at:

https://docs.amd.com/bundle/ROCm-Release-Notes-v5.3/page/About_This_Document.html

ROCm Documentation Team

**Comment by keryell** [comment]
Cool! Keep the momentum and get ready for Ubuntu 22.10 for the end of the month.

**Comment by xuhuisheng** [comment]
@keryell 
I am afraid rocm wont want to support non-lts version.
And ubuntu-22.10 upgraded kernel from 5.15 to 5.19, the amdgpu-dkms may not support this new version of linux kernel.

In my opinion, lts is enough.

**Comment by achede22** [comment]
@Bengt Please close the issue, this is SOLVED!

https://docs.amd.com/bundle/ROCm-Release-Notes-v5.3/page/About_This_Document.html

**Comment by Bengt** [comment]
Closing this issue due to recent releases now supporting Ubuntu 22.04.

**Comment by flowluap** [comment]
> It's release notes, which are great, but it does not provide a direct link to the downloads or installation guide. If anyone else sees this, go here instead https://docs.amd.com/bundle/ROCm-Downloads-Guide-v5.3/page/Introduction_to_AMD_ROCm_Installation_Downloads_Guide_for_Linux.html

Link is down

**Comment by Rmalavally** [comment]
@flowluap Thank you for letting us know. You can access the active link for the ROCm Downloads Guide at:

https://docs.amd.com/bundle/ROCm-Downloads-Guide-v5.3/page/Introduction_to_ROCm_Installation_Downloads_Guide_for_Linux.html

Here are some recommended ways to access the release documentation:
- Click Release Documentation for a consolidated list of all ROCm releases and available documentation
- Click the ROCm v5.x link for 5.x release documentation

Please let us know if you cannot access the ROCm Downloads Guide, and I can send you a PDF version (download from the docs portal). 

ROCm Documentation Team

**Comment by flowluap** [comment]
> @flowluap Thank you for letting us know. You can access the active link for the ROCm Downloads Guide at:
> 
> https://docs.amd.com/bundle/ROCm-Downloads-Guide-v5.3/page/Introduction_to_ROCm_Installation_Downloads_Guide_for_Linux.html
> 
> Here are some recommended ways to access the release documentation:
> 
> * Click Release Documentation for a consolidated list of all ROCm releases and available documentation
> * Click the ROCm v5.x link for 5.x release documentation
> 
> Please let us know if you cannot access the ROCm Downloads Guide, and I can send you a PDF version (download from the docs portal).
> 
> ROCm Documentation Team

@Rmalavally the link you sent works fine, thanks!
@Chase-san 's Link was down...
